setup/base.go

```diff
@@ -311,6 +312,18 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 		}
 	}
 
+	b.InternalAPIMux.HandleFunc("/health", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(200)
+	})
+
+	externalRouter.Handle("/health", httputil.MakeInternalAPI("Health", func(req *http.Request) util.JSONResponse {
```

Comment:
I just have these 2 attempts at handling `/health` to see what works (neither currently)

What's the distinction/difference between `internalRouter` and `externalRouter`?

Looking further, it seems like `b.InternalAPIMux` feeds into `internalRouter`.

---

Which addresses go where?

```
$  ./build.sh && ./bin/dendrite-demo-libp2p --port 8085
...
Our addresses: [/ip4/192.168.1.118/tcp/57741 /ip4/127.0.0.1/tcp/57741 /ip6/::1/tcp/57742]
```

 - http://localhost:8085/health -> 404 page not found
 - http://localhost:57741/health ->ERR_INVALID_HTTP_RESPONSE (same for below)
 - http://192.168.1.118:57741/health
 - http://127.0.0.1:57741/health
 - http://localhost:57742/health
 - http://192.168.1.118:57742/health
 - http://127.0.0.1:57742/health

Reply:
In monolith mode, the internal and external routers can end up being the same thing, but in full API mode or polylith mode, internal APIs go on the internal router and world-facing Matrix APIs (like the CS API and SS API) go on the external router.

None of those libp2p addresses will accept HTTP - those are for libp2p peering - you'll need to check port 8085, if that's what you specified with `--port`.

Reply:
Thanks @neilalexander!

I think my main confusion is that `setup/base.go` -> `SetupAndServeHTTP` isn't called in `cmd/dendrite-demo-libp2p/main.go` like it is for the monolith and polylith binaries so my health check wasn't being configured.